towhee/dataframe/dataframe.py

```diff
@@ -152,6 +155,7 @@ def get(self, start: int, count: int, block: bool = False) -> List[Tuple[Variabl
 
     def notify_block_readers(self):
         with self._accessible_cv:
+            self._kill_cv = True
```

Comment:
temporary fix, if block readers are killed then that dataframe wont be able to block read again. Only way to fix this is to make quite a few changes to df because we are limited to one conditional and that conditional needs to be able to tell between getting new values and being force woken up.